tests/remote/test_remote_encode.py

```diff
@@ -0,0 +1,222 @@
+# coding=utf-8
+# Copyright 2025 HuggingFace Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import PIL.Image
+import torch
+
+from diffusers.utils import load_image
+from diffusers.utils.remote_utils import (
+    DECODE_ENDPOINT_FLUX,
+    DECODE_ENDPOINT_SD_V1,
+    DECODE_ENDPOINT_SD_XL,
+    ENCODE_ENDPOINT_FLUX,
+    ENCODE_ENDPOINT_SD_V1,
+    ENCODE_ENDPOINT_SD_XL,
+    remote_decode,
+    remote_encode,
+)
+from diffusers.utils.testing_utils import (
+    enable_full_determinism,
+    slow,
+)
+
+
+enable_full_determinism()
+
+IMAGE = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/astronaut.jpg?download=true"
+
+
+class RemoteAutoencoderKLEncodeMixin:
+    channels: int = None
+    endpoint: str = None
+    decode_endpoint: str = None
+    dtype: torch.dtype = None
+    scaling_factor: float = None
+    shift_factor: float = None
+    image: PIL.Image.Image = None
+
+    def get_dummy_inputs(self):
+        if self.image is None:
+            self.image = load_image(IMAGE)
+        inputs = {
+            "endpoint": self.endpoint,
+            "image": self.image,
+            "scaling_factor": self.scaling_factor,
+            "shift_factor": self.shift_factor,
+        }
+        return inputs
+
+    def test_image_input(self):
+        inputs = self.get_dummy_inputs()
+        height, width = inputs["image"].height, inputs["image"].width
+        output = remote_encode(**inputs)
```

Comment:
I would run an assertion against a slice here, first in a separate test. And then have something like `test_image_input_encode_decode()` to test the full flow.

Reply:
VAE encode takes a random sample. We haven't exposed `deterministic` in `DiagonalGaussianDistribution` (never known it used?) or generator/seed for the sample (also never see it used), so the slice will likely change.

https://github.com/huggingface/diffusers/blob/4e3ddd5afab3a4b0b6265f210d6710933dade660/src/diffusers/models/autoencoders/vae.py#L691-L713

Reply:
> VAE encode takes a random sample. 

Not sure if I understand. We can call `encode()` on actual pixel values computed from an image:

https://github.com/huggingface/diffusers/blob/4e3ddd5afab3a4b0b6265f210d6710933dade660/examples/dreambooth/train_dreambooth_lora_sdxl.py#L1641

Reply:
Not `encode` specifically, the random sample is used in `latent_dist.sample` and controllable by `generator`, and `deterministic` is an option when creating `latent_dist`.

https://github.com/huggingface/diffusers/blob/4e3ddd5afab3a4b0b6265f210d6710933dade660/src/diffusers/models/autoencoders/vae.py#L704-L713

Reply:
Alright. Fine as is then.